source/common/upstream/health_checker_impl.cc

```diff
@@ -405,6 +405,22 @@ void TcpHealthCheckerImpl::TcpActiveHealthCheckSession::onEvent(uint32_t events)
       events & Network::ConnectionEvent::LocalClose) {
     parent_.dispatcher_.deferredDelete(std::move(client_));
   }
+
+  if ((events & Network::ConnectionEvent::Connected) && parent_.receive_bytes_.empty()) {
```

Comment:
Let me add a comment/TODO. There is no way with the current interfaces to actually make this work.